src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -522,7 +544,9 @@ bool SolverInterfaceImpl::isReadDataAvailable() const
   PRECICE_TRACE();
   PRECICE_CHECK(_state != State::Constructed, "initialize() has to be called before isReadDataAvailable().");
   PRECICE_CHECK(_state != State::Finalized, "isReadDataAvailable() cannot be called after finalize().");
-  return _couplingScheme->hasDataBeenReceived();
+  bool available = _couplingScheme->hasDataBeenReceived();
+  available |= (_allowsExperimental && _couplingScheme->hasInitialDataBeenReceived()); // if waveform relaxation is allowed, we always return true as soon as initialData is available.
```

Comment:
When merging this PR, all data always uses a Waveform. The important point here is that if a user uses the experimental interface, there will always be data available, because Waveforms are active and a user might want to sample. This is related to the discussion in https://github.com/precice/precice/issues/1223.

Here the experimental interface changes the behavior of an API function.

I would suggest to keep it as it is until #1223 is resolved. I can added link in a49d93302998be9f70c1f0cf9dded58551610841.

Reply:
I would suggest to check here against "order > 0".
"experimental" should really not switch on waveform relaxation, but the usage of an waveform order greater than one.
"experimental" should be the outer safeguard used by all experimental features.
Even in the long run (once waveform relaxation leaves the experimental stage), it should be possible to switch off waveform relaxation (for debugging or for accuracy tests etc.).

Reply:
I think you convinced me. I will try to find another solution. `order > 0` is at the current state probably the best solution. I still think that for consistency reasons a user might also want to sample at points inside the window, if the experimental interface is activated and `order == 0`, but this is quite academic and we should continue this discussion in #1223. For now `order > 0` is ok :+1:

Reply:
Done. 9aeeed7b9c66e94b20f3e2c46cbb781b1ca0cb4c